### PR TITLE
Allow units to be constructed from ValueTuples

### PR DESCRIPTION
**Pull request by tmilnthorp**
#699

### CODE REVIEW

**Comment by ZacharyPatten** [comment]
In my towel project I added a "MeasurementUnitSyntax" static class that had static fields for all the supported measurement units of all supported measurement types. Then I could add a "using static Towel.Measurements.MeasurementUnitSyntax" at the top of any file. That allows you to remove the enum type in the value tuple.

**Comment by ZacharyPatten** [comment]
Without the support of a "using static" it really isn't worth adding this feature because it doesn't change the syntax much.

File: UnitsNet.Tests/QuantityValueTests.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            QuantityValue decimalValue1 = 3.0m;
+            QuantityValue decimalValue2 = 3.0m;
+            QuantityValue decimalValue3 = 5.0m;
+
+            Assert.True(decimalValue1.Equals(decimalValue2));
+            Assert.False(doubleValue1.Equals(decimalValue3));
+        }
+
+        [Fact]
+        public void GetHashCodeEqualsValueGetHashCode()
```
Comment (angularsen) [inline UnitsNet.Tests/QuantityValueTests.cs]: These are nice tests!

File: UnitsNet.Tests/GeneratedCode/LengthTestsBase.g.cs [hunk 2]
```diff
@@ -549,5 +549,14 @@ public void BaseDimensionsShouldNeverBeNull()
         {
             Assert.False(Length.BaseDimensions is null);
         }
+
+        [Fact]
+        public void CanCreateInstanceFromTuple()
+        {
+            Length instance = (3.0, LengthUnit.Meter);
```
Comment (angularsen) [inline UnitsNet.Tests/GeneratedCode/LengthTestsBase.g.cs]: Can you add this code example to the README for how to use ValueTuple syntax.

**Comment by ZacharyPatten** [comment]
@angularsen I personally would wait on pulling this in until you have a static class you can use in an alias such as "using static SYNTAX." Example:

```csharp
using static Measurements.MeasurementSyntax; // <--- alias

namespace Example
{
  public static class MeasurementSyntax // <--- static class for alias
  {
    public static LengthUnit Meter = LengthUnit.Meter;
  }

  class Program
  {
    static void Main(string[] args)
    {
      Length length = (3.0, Meter); // <--- gets "Meter" from alias
    }
  }  
}
```

Without the static class to allow for an alias, the syntax isn't any cleaner IMO (and is probably worse than the current). However, with the static using, I think that is pretty clean as "Length" does not appear multiple times. It is just a bit of syntax sugar. Is it worth it? IDK... up to you...

**Pull request closed without merging**